tenseal/cpp/tensors/ckkstensor.h

```diff
@@ -79,6 +79,36 @@ class CKKSTensor : public EncryptedTensor<double, shared_ptr<CKKSTensor>>,
     vector<size_t> shape() const;
     shared_ptr<CKKSTensor> reshape(const vector<size_t>& new_shape);
     shared_ptr<CKKSTensor> reshape_inplace(const vector<size_t>& new_shape);
+
+    template <class T>
+    shared_ptr<T> broadcast_or_throw(const shared_ptr<T>& other) {
+        auto this_flat_size = this->_data.flat_size();
+        auto other_flat_size = other->_data.flat_size();
+
+        if (this_flat_size < other_flat_size) {
+            this->broadcast_inplace(other->shape());
+            return other;
+        } else {
+            return other->broadcast(this->shape());
+        }
+    }
+
+    template <class T>
+    T broadcast_or_throw(const T& other) {
+        auto this_flat_size = this->_data.flat_size();
+        auto other_flat_size = other.flat_size();
+
+        if (this_flat_size < other_flat_size) {
+            this->broadcast_inplace(other.shape());
+            return other;
+        } else {
+            return other.broadcast(this->shape());
+        }
+    }
```

Comment:
I don't feel quite confident about having this mutate the object inplace, should we necessarily broadcast `this`?

Reply:
Each operation already has the inplace/copy variants.
The resulting tensor will have the broadcasted shape anyway, right?
that's why it is broadcasted inplace.

Reply:
oh yeah, make sense